aeneas/src/os/Linux.v3

```diff
@@ -142,32 +143,40 @@ class LinuxTarget extends Target {
 
 			sections.startStringSection(w);
 
-			sections.addFuncSymbol(w, ".entry", mach.entryStub);
-			if (mach.allocStub != null) sections.addFuncSymbol(w, ".alloc", mach.allocStub);
-			if (mach.signalStub != null) sections.addFuncSymbol(w, ".signal", mach.signalStub);
-
-			var buf = StringBuilder.new();
-			for (i < prog.ir.methods.length) {
-				var m = prog.ir.methods[i];
-				if (m.source == null) continue;
-				var addr = mach.methodMap[m];
-				if (addr == null || addr.absolute < 0) continue;
-				var entry = ElfSymbol.new();
-				sections.symbols.put(entry);
-				entry.st_info = ElfConst.STB_GLOBAL | ElfConst.STT_FUNC;
-				entry.st_name = w.pos - sections.str_start;
-				entry.st_value = addr.absolute;
-				entry.st_size = addr.size;
-				entry.st_index = i16.view(sections.text.index);
-				buf.reset();
-				m.source.renderLong(buf);
-				buf.out(w.putkv);
-				w.putb(0);
+			if (CLOptions.SYMBOLS.val) {
+				sections.addFuncSymbol(w, ".entry", mach.entryStub);
+				if (mach.allocStub != null) sections.addFuncSymbol(w, ".alloc", mach.allocStub);
+				if (mach.signalStub != null) sections.addFuncSymbol(w, ".signal", mach.signalStub);
+
+				var buf = StringBuilder.new();
+				for (i < prog.ir.methods.length) {
```

Comment:
Now that this is nested another level, it's probably a good idea for us to factor out a method `addMethodSymbol()`.